mobile/library/java/io/envoyproxy/envoymobile/engine/AndroidNetworkMonitor.java

```diff
@@ -73,42 +70,36 @@ private AndroidNetworkMonitor(Context context, EnvoyEngine envoyEngine) {
       return;
     }
 
-    this.envoyEngine = envoyEngine;
-
     connectivityManager =
         (ConnectivityManager)context.getSystemService(Context.CONNECTIVITY_SERVICE);
-    networkRequest = new NetworkRequest.Builder()
-                         .addCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
-                         .build();
+    NetworkRequest networkRequest = new NetworkRequest.Builder()
+                                        .addCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
+                                        .build();
 
-    networkCallback = new NetworkCallback() {
-      @Override
-      public void onAvailable(Network network) {
-        handleNetworkChange();
-      }
+    NetworkCallback networkCallback = new NetworkCallback() {
       @Override
-      public void onCapabilitiesChanged(Network network, NetworkCapabilities networkCapabilities) {
-        handleNetworkChange();
+      public void onAvailable(@NonNull Network network) {
+        envoyEngine.onNetworkAvailable();
       }
+
       @Override
-      public void onLosing(Network network, int maxMsToLive) {
-        handleNetworkChange();
+      public void onCapabilitiesChanged(@NonNull Network network,
```

Comment:
is this guaranteed to be invoked only when there was a change?  do we want to check that engine's current preferred network value before calling `setPreferredNetwork`?